pkg/server/server.go

```diff
@@ -0,0 +1,691 @@
+	buckets, err := s.storage.ListBuckets()
+	if err != nil {
+		s.errorResponse(w, "InternalError", err.Error(), http.StatusInternalServerError)
+		return
+	}
+	
+	type Bucket struct {
+		Name         string    `xml:"Name"`
+		CreationDate time.Time `xml:"CreationDate"`
+	}
+	
+	type Owner struct {
+		ID          string `xml:"ID"`
+		DisplayName string `xml:"DisplayName"`
+	}
+	
+	type ListAllMyBucketsResult struct {
+		XMLName xml.Name `xml:"ListAllMyBucketsResult"`
+		Owner   Owner    `xml:"Owner"`
+		Buckets struct {
+			Bucket []Bucket `xml:"Bucket"`
+		} `xml:"Buckets"`
+	}
+	
+	result := ListAllMyBucketsResult{
+		Owner: Owner{
+			ID:          "local-user",
+			DisplayName: "local-user",
+		},
+	}
+	
+	for _, b := range buckets {
+		result.Buckets.Bucket = append(result.Buckets.Bucket, Bucket{
+			Name:         b.Name,
+			CreationDate: b.CreationDate,
+		})
+	}
+	
+	s.xmlResponse(w, result, http.StatusOK)
+}
+
+// handleCreateBucket handles CreateBucket operation
+func (s *S3Server) handleCreateBucket(w http.ResponseWriter, r *http.Request, bucket string) {
+	err := s.storage.CreateBucket(bucket)
+	if err != nil {
+		if err == storage.ErrBucketAlreadyExists {
+			s.errorResponse(w, "BucketAlreadyExists", "Bucket already exists", http.StatusConflict)
+		} else {
+			s.errorResponse(w, "InternalError", err.Error(), http.StatusInternalServerError)
+		}
+		return
+	}
+	
+	w.WriteHeader(http.StatusOK)
+}
+
+// handleDeleteBucket handles DeleteBucket operation
+func (s *S3Server) handleDeleteBucket(w http.ResponseWriter, r *http.Request, bucket string) {
+	err := s.storage.DeleteBucket(bucket)
+	if err != nil {
+		if err == storage.ErrBucketNotFound {
+			s.errorResponse(w, "NoSuchBucket", "Bucket does not exist", http.StatusNotFound)
+		} else {
+			s.errorResponse(w, "InternalError", err.Error(), http.StatusInternalServerError)
+		}
+		return
+	}
+	
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// handleHeadBucket handles HeadBucket operation
+func (s *S3Server) handleHeadBucket(w http.ResponseWriter, r *http.Request, bucket string) {
+	if !s.storage.BucketExists(bucket) {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+	
+	w.WriteHeader(http.StatusOK)
+}
+
+// handleListObjects handles ListObjects operation
+func (s *S3Server) handleListObjects(w http.ResponseWriter, r *http.Request, bucket string) {
+	query := r.URL.Query()
+	prefix := query.Get("prefix")
+	delimiter := query.Get("delimiter")
+	maxKeys := 1000
+	if mk := query.Get("max-keys"); mk != "" {
+		if parsed, err := strconv.Atoi(mk); err == nil {
+			maxKeys = parsed
+		}
+	}
+	
+	objects, commonPrefixes, err := s.storage.ListObjects(bucket, prefix, delimiter, maxKeys)
+	if err != nil {
+		if err == storage.ErrBucketNotFound {
+			s.errorResponse(w, "NoSuchBucket", "Bucket does not exist", http.StatusNotFound)
+		} else {
+			s.errorResponse(w, "InternalError", err.Error(), http.StatusInternalServerError)
+		}
+		return
+	}
+	
+	type Contents struct {
```

Comment:
For these structures, we can just use `type typename = source.typename` to reference the definition in the aws sdk,  and the other modifications as well.

Reply:
A new package has been added to specifically define these structures